packages/aws-cdk/lib/docker.ts

```diff
@@ -64,9 +64,10 @@ export async function prepareContainerAsset(asset: ContainerImageAssetMetadataEn
       shell(['docker', 'push', qualifiedImageName]),
       shell(['docker', 'push', stableImageName]),
     ]);
-    debug(` 👑  Docker image for ${asset.path} pushed.`);
+    const stableRef = await getStableRef(ecr.repositoryUri, imageId);
+    debug(` 👑  Docker image for ${asset.path} pushed, will use ${stableRef} to refer to it.`);
     return [
-      { ParameterKey: asset.imageNameParameter, ParameterValue: `${ecr.repositoryName}:${tag}` },
+      { ParameterKey: asset.imageNameParameter, ParameterValue: `${ecr.repositoryName}:${tag},${stableRef}` },
```

Comment:
Why do you need to pass both the `tag` and the `stableRef` here? The `tag` has already been pushed to ECR and for the task definition we only need `stableRef`, no?

Reply:
This is kind of a compromis to support all use cases on the `DockerImageAsset` API:
- `DockerImageAsset.imageUri` -> `repositoryUri:tag` ideal to publish images for Kubernetes resources (Pods) not created by CDK (CloudFormation) as (thanks to `repositoryName` and `imageTag` in `DockerImageAssetProps`) the name is well known and will not change.
- `DockerImageAsset.imageDigestUri` -> `repositoryUri@repositoryDigest` ideal for ECS/Fargate task also created by CDK (CloudFormation).